### PR TITLE
[main -> lts]: Fix/property dds/summarization bug

### PR DESCRIPTION
**Pull request by scottn12**
Port https://github.com/microsoft/FluidFramework/pull/13208 from main to lts branch.

This PR introduce a bug fix for PropertyDDS summarizer, where users
could end-up with inconsistent state in a very spefic scenario described
in more details in the [test](https://github.com/microsoft/FluidFramework/compare/main...nedalhy:FluidFramework:fix/property_dds/summarization_bug?expand=1#diff-4e672dc85e3021cf5cf96522a3bcf6653dfaaf10639ef2bf0c3de0a12fc8870eR91).

Also ports https://github.com/microsoft/FluidFramework/pull/11144 to get new test-utils functions for added unit tests

### CODE REVIEW

**Comment by DLehenbauer** [comment]
@ruiterr - FYI - We're cherry-picking your fix for Property DDS summaries into our LTS branch.

File: packages/runtime/runtime-utils/src/summarizerNode/summarizerNode.ts [hunk 1]
```diff
@@ -286,7 +286,12 @@ export class SummarizerNode implements IRootSummarizerNode {
         );
         return { latestSummaryUpdated: true, wasSummaryTracked: false, snapshot: snapshotTree };
     }
-
+/**
```
Comment (ssimic2) [inline packages/runtime/runtime-utils/src/summarizerNode/summarizerNode.ts:289]: Please check with @NicholasCouri if its safe to pick this change here.

**Comment by ssimic2** [comment]
Also please make sure that original owner of the change has seen/approved this PR.

**Review by ssimic2 — APPROVED** [review]

File: packages/runtime/runtime-utils/src/summarizerNode/summarizerNode.ts (hunk 1 above)
Comment (NicholasCouri) [reply packages/runtime/runtime-utils/src/summarizerNode/summarizerNode.ts:289]: looks good to me :)

**Comment by ruiterr** [comment]
@scottn12 There are quite a few changes in this PR that are unrelated to our original PR. The original change only affected three files 
experimental/PropertyDDS/packages/property-dds/package.json
experimental/PropertyDDS/packages/property-dds/src/propertyTree.ts
experimental/PropertyDDS/packages/property-dds/src/test/propertyTree.spec.ts

are the other changes intentionally included? Were those changes that are needed for our modifications to work?

**Comment by ruiterr** [comment]
The changes in the three files above LGTM. However, I can't really comment on the impact of the oher included files.

**Comment by scottn12** [comment]
> are the other changes intentionally included? Were those changes that are needed for our modifications to work?
> 
> Are all these changes only needed because we wrote a test that depends on `@fluidframework/test-drivers:^1.4.0`?

@ruiterr, yes that is correct. But it was actually due to the writing a test that relies on `@fluidframework/test-utils`. Specifically, the `createSummarizer` and `summarizeNow` functions.

**Comment by ruiterr** [comment]
I see. Thanks for confirming. Then I assume that adding those files is necessary, unless we would want to leave out the test. The actual change should not require any additional dependencies.

**Pull request merged**